### PR TITLE
Add native/firstParty flag to cpp and dotnet instrumentation

### PR DESCRIPTION
**Pull request by haidong**
To address #4795, this is a series of PRs that add native or first party flags to registries that lack them. This batch touches cpp and dotnet instrumentation registries.

Entries whose repo url returns 404 are deleted.

PR https://github.com/open-telemetry/opentelemetry.io/pull/4836 has more background.

### CODE REVIEW

File: data/registry/instrumentation-cpp-httpd.yml [hunk 1]
```diff
@@ -13,3 +13,4 @@ authors:
 urls:
   repo: https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/httpd
 createdAt: 2021-11-11
+isFirstParty: true
```
Comment (svrnm) [inline data/registry/instrumentation-cpp-httpd.yml:16]:
```suggestion
isFirstParty: false
```

File: data/registry/instrumentation-cpp-nginx.yml [hunk 2]
```diff
@@ -14,3 +14,4 @@ authors:
 urls:
   repo: https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/nginx
 createdAt: 2021-11-11
+isFirstParty: true
```
Comment (svrnm) [inline data/registry/instrumentation-cpp-nginx.yml:17]:
```suggestion
isFirstParty: false
```

**Review by svrnm — CHANGES_REQUESTED** [review]
All of these components are non first party, as they are not created by the same organisation that has created the underlying instrumented library, so can you switch from "true" to "false" whenever the lib lives in an otel repo

**Comment by haidong** [comment]
> Some of them have moved, and not been deleted, see this PR: #4822, I didn't merge this PR yet since I had limited availability last week, I will fix this now and then we can delete the ones remaining.

I have rebased my branch using upstream/main. I've also made the changes requested. Please review again when convenient. Thanks!

File: data/registry/instrumentation-dotnet-azure-sdk.yml [hunk 3]
```diff
@@ -15,3 +15,4 @@ authors:
 urls:
   repo: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md#opentelemetry-with-azure-monitor-zipkin-and-others
 createdAt: 2021-12-16
+isFirstParty: true
```
Comment (haidong) [inline data/registry/instrumentation-dotnet-azure-sdk.yml:18]: @svrnm This is from Azure, not open-telemetry, so I marked `isFirstParty` to `true`.

Comment (svrnm) [reply data/registry/instrumentation-dotnet-azure-sdk.yml:18]: yes, here it makes sense to set this as isFirstParty

File: data/registry/instrumentation-dotnet-kafkaflow.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ authors:
 urls:
   repo: https://github.com/Farfetch/kafkaflow/tree/master/src/KafkaFlow.OpenTelemetry
 createdAt: 2023-12-07
-isFirstParty: true
+isFirstParty: false
```
Comment (svrnm) [inline data/registry/instrumentation-dotnet-kafkaflow.yml:19]: Can you revert this change? This is firstParty

Comment (haidong) [reply data/registry/instrumentation-dotnet-kafkaflow.yml]: Done!

**Review by svrnm — APPROVED** [review]
